R/reportPrices.R

```diff
@@ -31,134 +31,164 @@
-                             "Price|Final Energy|Transport|Gases (US$2005/GJ)"))
-    }
+  demSe          <- readGDX(gdx,name=c("vm_demSe","v_demSe"),
+                            types="variables",field="l",format="first_found",
+                            restore_zeros=FALSE)[,t,]
+
+  tdptwyr2dpgj <- 31.71   # conversion factor trUSD2005/TWa to USD2005/GJ
+
+  ## weights for market aggregtion of prices: FE share of market
+  p_weights_FEprice_mkt <- dimSums(vm_demFeSector, dim=3.1, na.rm = T) / dimSums(vm_demFeSector, dim=c(3.1,3.4), na.rm = T)
+  p_weights_FEprice_mkt[is.na(p_weights_FEprice_mkt)] <- 0
+  ## adjust to pm_FEprice dimensions
+  p_weights_FEprice_mkt <- p_weights_FEprice_mkt[,getYears(pm_FEPrice),getNames(pm_FEPrice)]
+
+  ## weights for fepet/fedie to liquids aggregtion of prices: FE fepet/fedie share of aggregated markets
+  p_weights_FEprice_diepet <- dimSums(mselect(vm_demFeSector, all_enty1=c("fedie","fepet"), emi_sectors="trans"), dim=c(3.1,3.4), na.rm = T) /
+    dimSums(mselect(vm_demFeSector, all_enty1=c("fedie","fepet"), emi_sectors="trans"), dim=c(3.1,3.2,3.4), na.rm = T)
+  p_weights_FEprice_diepet[is.na(p_weights_FEprice_diepet)] <- 0
+  p_weights_FEprice_diepet <- p_weights_FEprice_diepet[,getYears(pm_FEPrice),]
+
+  out <- NULL
+
+  ## calculate prices as weighted average prices across markets
+  ## FE Transport Prices
+  out <- mbind(
+    out,
+    setNames( dimSums(mselect(p_weights_FEprice_mkt * pm_FEPrice, all_enty1="feelt", emi_sectors = "trans"), dim=3.3, na.rm = T)*tdptwyr2dpgj,
+             "Price|Final Energy|Transport|Electricity (US$2005/GJ)"),
+    ## in case of transport liquids: calculate weighted average of markets first, then calculate weighted average of fepet/fedie
+    setNames( dimSums( p_weights_FEprice_diepet * dimSums(mselect(p_weights_FEprice_mkt * pm_FEPrice, all_enty1=c("fepet","fedie"), emi_sectors = "trans"), dim=3.3, na.rm = T), dim=3.1, na.rm = T)*tdptwyr2dpgj,
+             "Price|Final Energy|Transport|Liquids (US$2005/GJ)"),
+    setNames( dimSums(mselect(p_weights_FEprice_mkt * pm_FEPrice, all_enty1="feh2t", emi_sectors = "trans"), dim=3.3, na.rm = T)*tdptwyr2dpgj,
+             "Price|Final Energy|Transport|Hydrogen (US$2005/GJ)")
+  )
+
+  if (module2realisation["transport",2] == "edge_esm") {
+    out <- mbind(
+      out,
+      setNames( dimSums(mselect(p_weights_FEprice_mkt * pm_FEPrice, all_enty1="fegat", emi_sectors = "trans"), dim=3.3, na.rm = T)*tdptwyr2dpgj,
+               "Price|Final Energy|Transport|Gases (US$2005/GJ)"))
+  }
+
+  out <- mbind(
+    out,
+    setNames( dimSums(mselect(p_weights_FEprice_mkt * pm_FEPrice, all_enty1="fedie", emi_sectors = "trans"), dim=3.3, na.rm = T)*tdptwyr2dpgj,
+             "Price|Final Energy|Transport|Liquids|HDV (US$2005/GJ)"),
+    setNames( dimSums(mselect(p_weights_FEprice_mkt * pm_FEPrice, all_enty1="fepet", emi_sectors = "trans"), dim=3.3, na.rm = T)*tdptwyr2dpgj,
```

Comment:
Please change the name according to the new naming convention to ``Price|Final Energy|Transport|LDV|Liquids``. We now want to do all sectors first, then energy carriers, then energy sources or energy uses.

Reply:
yeah I thought about this. The problem I see here is that these are essentially also different types of liquids. So in this sense it follow your logic already. It is also confusing, if you take HDV/LDV as mode type, that the same distinction is not made for other carriers (there is no `HDV|Electric` and the like).

Reply:
It also does not make sense to add `HDV|Electric` since the price signal is equivalent.